### PR TITLE
TESB416-246 Added scraper for schauburg and fixed some bugs

### PR DESCRIPTION
**Pull request by flitzpiepe96**
Basically the title says everything

### CODE REVIEW

File: data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/KinemathekScraper.java [hunk 1]
```diff
@@ -75,7 +75,7 @@ private void handleMovies(List<WebElement> movieElements, LocalDate date) {
             Playlist playlist = new Playlist();
             playlist.setCinema(cinema);
             playlist.setMovie(movie);
-            playlist.setTime(Date.from(dateTime.atZone(ZoneId.of("GMT+1")).toInstant()));
+            playlist.setTime(Date.from(dateTime.atZone(ZoneId.of("GMT+2")).toInstant()));
```
Comment (rgcjonas) [inline data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/KinemathekScraper.java]: Use `ZoneId.of("Europe/Berlin")` to automatically account for DST

(Yeah, next semester case, but...)

File: data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/KurbelScraper.java [hunk 2]
```diff
@@ -68,7 +68,7 @@ private void handlePlaytimes(Movie movie, WebElement parent) {
                 Playlist playlist = new Playlist();
                 playlist.setCinema(cinema);
                 playlist.setMovie(movie);
-                playlist.setTime(Date.from(dateTime.atZone(ZoneId.of("GMT+1")).toInstant()));
+                playlist.setTime(Date.from(dateTime.atZone(ZoneId.of("GMT+2")).toInstant()));
```
Comment (rgcjonas) [inline data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/KurbelScraper.java]: same thing

File: data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/SchauburgScraper.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+                String title = movieRow.findElement(By.xpath(".//td[2]/a")).getText();
+                LocalDateTime dateTime = addTimeToDate(timeText, dates.get(movieElements.indexOf(movieElement)));
+
+                Movie movie = new Movie(title, null, null, 0, null, null);
+                movies.add(movie);
+
+                Playlist playlist = new Playlist();
+                playlist.setCinema(cinema);
+                playlist.setMovie(movie);
+                playlist.setTime(Date.from(dateTime.atZone(ZoneId.of("GMT+2")).toInstant()));
```
Comment (rgcjonas) [inline data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/SchauburgScraper.java]: .

File: data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/AbstractCinemaScraper.java [hunk 4]
```diff
@@ -40,39 +40,46 @@
 
     public AbstractCinemaScraper() {
         tmdb = new TmdbDataRetriever();
+        cinema = new Cinema();
+        cinema.setName("UNKNOWN");
```
Comment (rgcjonas) [inline data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/AbstractCinemaScraper.java]: Do we really want this? I believe we should rather crash and burn if the subclass doesn't set a cinema

File: data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/AbstractCinemaScraper.java [hunk 5]
```diff
@@ -40,39 +40,46 @@
...
 
     public void scrape() {
         logger = getLogger();
-        cinema = saveObject(getCinema(), Cinema.class);
-
         logger.info(String.format("Initializing Webdriver for scraper: [%s]", cinema.getName()));
         initlializeWebdriver();
 
+        cinema = saveObject(getCinema(), Cinema.class);
+
```
Comment (rgcjonas) [inline data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/AbstractCinemaScraper.java]: Is there a specific reason for moving it around? Just wondering

File: data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/AbstractCinemaScraper.java [hunk 6]
```diff
@@ -143,7 +151,8 @@ private Movie retrieveMovieInformation(Movie movie) {
         try {
             return tmdb.getMovie(movie);
         } catch (MovieDbException e) {
-            //Movie DB does not know this movie for some reason. Basically this means we won't show it
+            // Movie DB does not know this movie for some reason. Basically this
+            // means we won't show it
```
Comment (rgcjonas) [inline data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/AbstractCinemaScraper.java]: This smells like unnecessary change, but I think I'm going to allow it because the space at the beginning makes it prettier

File: data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/KinemathekScraper.java (hunk 1 above)
Comment (flitzpiepe96) [inline data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/KinemathekScraper.java]: I actually didnt know there is something like that. I am changing that immediatelly ^^

File: data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/AbstractCinemaScraper.java (hunk 4 above)
Comment (flitzpiepe96) [inline data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/AbstractCinemaScraper.java]: Problem is that you use the web driver in zkm scraper in the getCinema method, so the driver has to be initialized before. But then we have the problem that initializing the driver uses the cinema name (its just for logging, but I didnt feel like inserting null checks at a thousand lines of code). This way the first few logging entries are unknown, but after the cinema is set correctly there are no problems anymore. 

Nevertheless your right and I will insert a null check after the getCinema method. If it is null we will throw an exception instead of continuing with unknown.

File: data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/AbstractCinemaScraper.java (hunk 5 above)
Comment (flitzpiepe96) [inline data retrieval/src/main/java/de/tinf15b4/kino/retrieval/scraper/AbstractCinemaScraper.java]: Yes there is. See comment above
